src/controller.cpp

```diff
@@ -1135,32 +1137,54 @@ void controller::enqueue_items(std::shared_ptr<rss_feed> feed)
 }
 
 std::string controller::generate_enqueue_filename(const std::string& url,
+	const std::string& title,
+	const time_t pubDate,
 	std::shared_ptr<rss_feed> feed)
 {
 	std::string dlformat = cfg.get_configvalue("download-path");
 	if (dlformat[dlformat.length() - 1] != NEWSBEUTER_PATH_SEP[0])
 		dlformat.append(NEWSBEUTER_PATH_SEP);
 
-	fmtstr_formatter fmt;
-	fmt.register_fmt('n', feed->title());
-	fmt.register_fmt('h', get_hostname_from_url(url));
+	std::string filemask = cfg.get_configvalue("download-filename");
+	dlformat.append(filemask);
 
-	std::string dlpath = fmt.do_format(dlformat);
+	auto time_formatter = [&pubDate](const char* format) {
+		char pubDate_formatted[1024];
+		strftime(pubDate_formatted,
+				sizeof(pubDate_formatted),
+				format,
+				localtime(&pubDate));
+		return  std::string(pubDate_formatted);
+	};
 
 	char buf[2048];
 	snprintf(buf, sizeof(buf), "%s", url.c_str());
 	char* base = basename(buf);
-	if (!base || strlen(base) == 0) {
-		char lbuf[128];
-		time_t t = time(nullptr);
-		strftime(lbuf,
-			sizeof(lbuf),
-			"%Y-%b-%d-%H%M%S.unknown",
-			localtime(&t));
-		dlpath.append(lbuf);
-	} else {
-		dlpath.append(base);
+	std::string extension;
+	if (base) {
+		std::string base_s (base);
+		std::size_t pos = base_s.rfind('.');
+		if (pos != std::string::npos)
+			extension.append(base_s.substr(pos+1));
 	}
+
+	fmtstr_formatter fmt;
+	fmt.register_fmt('n', feed->title());
+	fmt.register_fmt('h', get_hostname_from_url(url));
+	fmt.register_fmt('u', base);
+	fmt.register_fmt('F', time_formatter(_("%F")));
```

Comment:
Forgot to comment on that, sorry: please remove `_` from all of the calls to `register_fmt`. Placeholders aren't meant to be localized.

Reply:
No problem ;)